src/jit/lowerarmarch.cpp

```diff
@@ -280,93 +253,68 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
             // it to a larger constant whose size is sufficient to support
             // the largest width store of the desired inline expansion.
 
-            ssize_t fill = initVal->gtIntCon.gtIconVal & 0xFF;
+            ssize_t fill = src->AsIntCon()->IconValue() & 0xFF;
+
             if (fill == 0)
             {
-                MakeSrcContained(blkNode, source);
+                src->SetContained();
             }
-            else if (size < REGSIZE_BYTES)
+            else if (size >= REGSIZE_BYTES)
             {
-                initVal->gtIntCon.gtIconVal = 0x01010101 * fill;
+                fill *= 0x0101010101010101LL;
+                src->gtType = TYP_LONG;
             }
             else
             {
-                initVal->gtIntCon.gtIconVal = 0x0101010101010101LL * fill;
-                initVal->gtType             = TYP_LONG;
+                fill *= 0x01010101;
             }
-            blkNode->gtBlkOpKind = GenTreeBlk::BlkOpKindUnroll;
+
+            src->AsIntCon()->SetIconValue(fill);
```

Comment:
The value wasn't updated when it was 0, even though the value that was checked for wasn't the original value but a masked one...

Reply:
Umm, I changed because it simply was confusing - lowering was masking and checking for 0 and codegen was NOT masking but also checking for 0. But yes, it turns out that it is a real bug on ARM:
```
        52822000          movz    w0, #0x1100
        72A00020          movk    w0, #1 LSL #16
        9101C3A1          add     x1, fp, #112  // [V05 loc0]
        B9000020          str     w0, [x1]
        39001020          strb    w0, [x1,#4]
```
XARCH lowering was correct but also did things differently - masking, multiplying & updating, checking for 0.